apps/client/components/ShopifyPurchaseOrdersTable.tsx

```diff
@@ -0,0 +1,266 @@
+    plural: 'orders',
+  };
+
+  const [sortDirection, setSortDirection] = useState<IndexTableSortDirection>();
+  const [sortedColumnIndex, setSortedColumnIndex] = useState<number>();
+
+  const handleSort = (
+    headingIndex: number,
+    direction: IndexTableSortDirection
+  ) => {
+    setSortDirection(direction);
+    setSortedColumnIndex(headingIndex);
+    const headingKey = [
+      'id',
+      'vendor_name',
+      'order_date',
+      'expected_delivery_date',
+      'quantity',
+      'totalCost',
+    ][headingIndex] as keyof ComputedPurchaseOrdersRow;
+    const sortedRows = [...computedPurchaseOrdersRows].sort((rowA, rowB) => {
+      const valueA = rowA[headingKey];
+      const valueB = rowB[headingKey];
+      if (typeof valueA === 'string') {
+        return valueA.localeCompare(valueB as string);
+      }
+      if (typeof valueA === 'number' && typeof valueB === 'number') {
+        return valueA - valueB;
+      }
+      if (valueA instanceof Date && valueB instanceof Date) {
+        return valueA.getTime() - valueB.getTime();
+      }
+      return 0;
+    });
+    if (direction === 'descending') {
+      sortedRows.reverse();
+    }
+    setComputedPurchaseOrdersRows(sortedRows);
+  };
+
+  const { selectedResources, allResourcesSelected, handleSelectionChange } =
+    useIndexResourceState(computedPurchaseOrdersRows);
+
+  const rowMarkup = useMemo(
+    () =>
+      computedPurchaseOrdersRows.map(
+        (
+          {
+            id,
+            vendor_name,
+            order_date,
+            expected_delivery_date,
+            quantity,
+            totalCost,
+            purchase_order_line_items,
+          },
+          index
+        ) => {
+          console.log(selectedResources);
+          const selected = selectedResources.includes(id.toString());
+          return (
+            <>
+              <IndexTable.Row
+                id={id.toString()}
+                key={id}
+                selected={selected}
+                position={index}
+              >
+                <IndexTable.Cell>
+                  <Text as="span">{id}</Text>
+                </IndexTable.Cell>
+                <IndexTable.Cell>{vendor_name}</IndexTable.Cell>
+                <IndexTable.Cell>
+                  {' '}
+                  <Text as="span" alignment="end" numeric>
+                    {dateFormat(order_date)}
+                  </Text>
+                </IndexTable.Cell>
+                <IndexTable.Cell>
+                  {' '}
+                  <Text as="span" alignment="end" numeric>
+                    {dateFormat(expected_delivery_date)}
+                  </Text>
+                </IndexTable.Cell>
+
+                <IndexTable.Cell>
+                  <Text as="span" alignment="end" numeric>
+                    {quantity}
+                  </Text>
+                </IndexTable.Cell>
+                <IndexTable.Cell>
+                  <Text as="span" alignment="end" numeric>
```

Comment:
Did you consider adding components for each datatype that would handle `date`, `numeric`, `string` data and apply the necessary formatting consistently